vendor/engines/saas/lib/saas/engine.rb

```diff
@@ -17,6 +17,19 @@ class Engine < ::Rails::Engine
         })
       end
 
+      LoginController.class_eval do
+        before_action :page_job, only: [:public, :private]
+
+        :private
+
+        def page_job
+          if request.referer !~ /github\.com/
```

Comment:
This seems kind of brittle - what if we checked for one of the OAuth parameters?